### PR TITLE
Functional tests - Product settings enable/disable display discounted price

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Enable/Disable display discounted price
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18207)

### CODE REVIEW

File: tests/puppeteer/pages/FO/product.js [hunk 1]
```diff
@@ -124,4 +126,22 @@ module.exports = class Product extends FOBasePage {
...
+  }
+
+  /**
+   * Is unit discount visible
+   * @returns {Promise<boolean>}
+   */
+  async isUnitDiscountVisible() {
+    const isUnitDiscount = (await this.getTextContent(this.discountColumn) === 'Unit discount');
+    return isUnitDiscount;
+  }
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/FO/product.js]:
```suggestion
   * Get discount value
   * @returns {string}
   */
  async getDiscountValue() {
    return this.getTextContent(this.discountColumn);
  }
```

File: tests/puppeteer/pages/BO/catalog/discounts/catalogPriceRules/index.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  /**
+   * Delete catalog price rule
+   * @param ruleName
+   * @returns {Promise<string>}
+   */
+  async deleteCatalogPriceRule(ruleName) {
+    await this.dialogListener(true);
+    if (await this.elementVisible(this.filterNameColumn)) await this.filterTableByRuleName(ruleName);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/catalog/discounts/catalogPriceRules/index.js]:
```suggestion
    if (await this.elementVisible(this.filterNameColumn)) {
      await this.filterTableByRuleName(ruleName);
    }
```

File: tests/puppeteer/pages/BO/catalog/discounts/index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    this.catalogPriceRulesTab = '#subtab-AdminSpecificPriceRule';
+  }
+
+  /* Methods */
+  /**
+   * Change Tab to Catalog Price Rules in Discounts Page
+   * @returns {Promise<void>}
+   */
+  async goToCatalogPriceRulesTab() {
+    await this.page.click(this.catalogPriceRulesTab);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/catalog/discounts/index.js]:
```suggestion
    await this.clickAndWaitForNavigation(this.catalogPriceRulesTab);
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  // Login into BO and go to products page
+  loginCommon.loginBO();
+
+  it('should go to \'Catalog > Discounts\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToDiscountsPage', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.catalogParentLink,
+      this.pageObjects.boBasePage.discountsLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js]:
```suggestion
```

not a migrated page, you can delete this

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      this.pageObjects.boBasePage.shopParametersParentLink,
+      this.pageObjects.boBasePage.productSettingsLink,
+    );
+    const pageTitle = await this.pageObjects.productSettingsPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.productSettingsPage.pageTitle);
+  });
+
+  const tests = [
+    {args: {action: 'disable', unitPrice: false, unitDiscount: true}},
+    {args: {action: 'enable', unitPrice: true, unitDiscount: false}},
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js]:
```suggestion
    {args: {action: 'disable', unitDiscount: true, textToCheck: 'Unit price'}},
    {args: {action: 'enable', unitPrice: true, textToCheck: 'Unit discount'}},
```

No need to add false values

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        `${test.args.action}DisplayDiscountedPrice`,
+        baseContext,
+      );
+      const result = await this.pageObjects.productSettingsPage.setDisplayDiscountedPriceStatus(
+        test.args.unitPrice,
+      );
+      await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
+    });
+
+    it('should check the existence of the \'Unit discount\' or the \'Unit price\'', async function () {
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js]:
```suggestion
    it('should check the existence of the Unit value'', async function () {
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+        baseContext,
+      );
+      page = await this.pageObjects.productSettingsPage.viewMyShop();
+      this.pageObjects = await init();
+      await this.pageObjects.homePage.changeLanguage('en');
+      await this.pageObjects.homePage.goToProductPage(1);
+      const isUnitPriceVisible = await this.pageObjects.productPage.isUnitPriceVisible();
+      await expect(isUnitPriceVisible).to.equal(test.args.unitPrice);
+      const isUnitDiscountVisible = await this.pageObjects.productPage.isUnitDiscountVisible();
+      await expect(isUnitDiscountVisible).to.equal(test.args.unitDiscount);
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js]:
```suggestion
      const disountValue = this.pageObjects.productPage.getDiscountValue();
      await expect(discountValue).to.equal(test.args.textToCheck);
```

Following the suggestion on FO/product.js file

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js (hunk 4 above)
Comment (Progi1984) [reply tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js]: :+1:

Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js]: Done thx :)

**Review by boubkerbribri — DISMISSED** [review]

**Review by NeOMakinG — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by NeOMakinG** [comment]
@nesrineabdmouleh there are some conflicts that you need to resolve before merging :)

**Review by NeOMakinG — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

File: tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js [hunk 8]
```diff
@@ -53,4 +53,48 @@ module.exports = class customerSettings extends BOBasePage {
...
+  /**
+   * Enable/disable redisplay cart at login
+   * @param toEnable, true to enable and false to disable
+   * @return {Promise<string>}
+   */
+  async setRedisplayCartAtLoginStatus(toEnable = true) {
+    await this.waitForSelectorAndClick(this.redisplayCartAtLoginLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+    await this.clickAndWaitForNavigation(this.saveGeneralFormButton);
+    return this.getTextContent(this.alertSuccessBlock);
+  }
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js]:
```suggestion
```

File: tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js [hunk 9]
```diff
@@ -53,4 +53,48 @@ module.exports = class customerSettings extends BOBasePage {
...
+  /**
+   * Enable/Disable send email after registration
+   * @param toEnable
+   * @returns {Promise<string>}
+   */
+  async setEmailAfterRegistrationStatus(toEnable = true) {
+    await this.waitForSelectorAndClick(this.sendEmailAfterRegistrationLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+    await this.clickAndWaitForNavigation(this.saveGeneralFormButton);
+    return this.getTextContent(this.alertSuccessBlock);
+  }
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js]:
```suggestion
```

File: tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js [hunk 10]
```diff
@@ -53,4 +53,48 @@ module.exports = class customerSettings extends BOBasePage {
...
+  /**
+   * Enable/Disable ask for birth date
+   * @param toEnable
+   * @returns {Promise<string|*>}
+   */
+  async setAskForBirthDate(toEnable = true) {
+    await this.waitForSelectorAndClick(this.askForBirthDateLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+    await this.clickAndWaitForNavigation(this.saveGeneralFormButton);
+    return this.getTextContent(this.alertSuccessBlock);
+  }
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js]:
```suggestion
```

File: tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js [hunk 11]
```diff
@@ -53,4 +53,48 @@ module.exports = class customerSettings extends BOBasePage {
...
+  /**
+   * Enable/Disable partner offer
+   * @param toEnable
+   * @returns {Promise<string>}
+   */
+  async setPartnerOfferStatus(toEnable = true) {
+    await this.waitForSelectorAndClick(this.enablePartnerOfferLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+    await this.clickAndWaitForNavigation(this.saveGeneralFormButton);
+    return this.getTextContent(this.alertSuccessBlock);
+  }
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js]:
```suggestion
```
You don't need modification on this file. all these functions were deleted and replaced by `setOptionStatus`

Comment (nesrineabdmouleh) [reply tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js]: Done :)

**Review by Progi1984 — DISMISSED** [review]

**Review by NeOMakinG — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**